### PR TITLE
feat(cognito): OpenID Connect identity provider

### PR DESCRIPTION
**Pull request by jogold**
Add the `UserPoolIdentityProviderOidc` class to create an OpenID Connect
identity provider for user pools.

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/master/INTEGRATION_TESTS.md)?
	* [x] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-cognito/lib/user-pool-idps/oidc.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  public readonly providerName: string;
+
+  constructor(scope: Construct, id: string, props: UserPoolIdentityProviderOidcProps) {
+    super(scope, id, props);
+
+    const scopes = props.scopes ?? ['openid'];
+
+    const resource = new CfnUserPoolIdentityProvider(this, 'Resource', {
+      userPoolId: props.userPool.userPoolId,
+      providerName: props.name,
```
Comment (corymhall) [inline packages/@aws-cdk/aws-cognito/lib/user-pool-idps/oidc.ts:111]: Would it be better to use `physicalName` here and generate one if the user doesn't provide one? It doesn't look like OIDC has a specific name requirement like the others.

File: packages/@aws-cdk/aws-cognito/lib/user-pool-idps/oidc.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+   * The method to use to request attributes
+   *
+   * @default OidcAttributeRequestMethod.GET
+   */
+  readonly attributeRequestMethod?: OidcAttributeRequestMethod
+
+  /**
+   * Issuer URL
+   */
+  readonly issuerUrl: string;
```
Comment (corymhall) [inline packages/@aws-cdk/aws-cognito/lib/user-pool-idps/oidc.ts:52]: nit - can you move this up so required comes before optional?

**Review by corymhall — CHANGES_REQUESTED** [review]
Looks good @jogold! Just a couple of minor comments.

File: packages/@aws-cdk/aws-cognito/lib/user-pool-idps/oidc.ts (hunk 1 above)
Comment (corymhall) [reply packages/@aws-cdk/aws-cognito/lib/user-pool-idps/oidc.ts]: ahh I missed that it wasn't extending `Resource` directly. How about:

```ts
providerName: props.name || Lazy.string({ produce: () => Names.uniqueId(this) }),
```

Comment (jogold) [reply packages/@aws-cdk/aws-cognito/lib/user-pool-idps/oidc.ts]: `id` might be a better default since the provider name is limited to 32 chars?

**Review by corymhall — APPROVED** [review]

**Pull request merged**